### PR TITLE
[Exporter.Geneva] Refactor GenevaExporter

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Refactor GenevaTraceExporter and GenevaLogExporter to make their behavior configurable in the ctor
- This would allow us switch from MsgPack to TLD later on
- Moved the classes with extension methods outside of the `MsgPackExporter` folder

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs [hunk 1]
```diff
@@ -21,12 +21,11 @@
 using System.Runtime.InteropServices;
 using System.Threading;
 using Microsoft.Extensions.Logging;
-using OpenTelemetry.Internal;
 using OpenTelemetry.Logs;
 
 namespace OpenTelemetry.Exporter.Geneva;
 
-public class GenevaLogExporter : GenevaBaseExporter<LogRecord>
+internal sealed class MsgPackLogExporter : IDisposable
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs:28]: Curious - why isn't this derived from `GenevaBaseExporter<LogRecord>`?

Comment (utpilla) [reply src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs:28]: There is no strong reason. We could do that as well. I don't see any major benfits to doing that either. In fact, I think we could move all of those internal fields and methods of `GenevBaseExporter` to a new abstract class called `MsgPackExporter` as all of those internal data is related to to MsgPack approach.

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs:28]: Got it, makes sense 👍

File: src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs [hunk 2]
```diff
@@ -154,31 +150,6 @@ public override ExportResult Export(in Batch<LogRecord> batch)
...
-        {
-            return;
-        }
-
-        if (disposing)
-        {
-            // DO NOT Dispose m_buffer as it is a static type
-            try
-            {
-                (this.m_dataTransport as IDisposable)?.Dispose();
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs:169]: who is now responsible for disposing transport?

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs:169]: ignore.. i see now that the code just moved!

**Review by cijothomas — APPROVED** [review]

**Pull request merged**